server/service/integration_mdm_test.go

```diff
@@ -6794,6 +6794,60 @@ func (s *integrationMDMTestSuite) TestValidManagementRequestNoAuth() {
 	require.True(t, s.checkIfXMLTagContains("Data", pendingCmd.SettingValue, resMsg))
 }
 
+func (s *integrationMDMTestSuite) TestValidManagementUnenrollRequest() {
+	t := s.T()
+
+	// Target Endpoint URL for the management endpoint
+	targetEndpointURL := microsoft_mdm.MDE2ManagementPath
+
+	// Target Device ID
+	deviceID := "DB257C3A08778F4FB61E2749066C1F27"
+
+	enrolledDevice := &fleet.MDMWindowsEnrolledDevice{
+		MDMDeviceID:            deviceID,
+		MDMHardwareID:          uuid.New().String() + uuid.New().String(),
+		MDMDeviceState:         uuid.New().String(),
+		MDMDeviceType:          "CIMClient_Windows",
+		MDMDeviceName:          "DESKTOP-1C3ARC1",
+		MDMEnrollType:          "ProgrammaticEnrollment",
+		MDMEnrollUserID:        "",
+		MDMEnrollProtoVersion:  "5.0",
+		MDMEnrollClientVersion: "10.0.19045.2965",
+		MDMNotInOOBE:           false,
+	}
+
+	err := s.ds.MDMWindowsInsertEnrolledDevice(context.Background(), enrolledDevice)
+	require.NoError(t, err)
+
+	// Checking if device was enrolled
+	_, err = s.ds.MDMWindowsGetEnrolledDeviceWithDeviceID(context.Background(), deviceID)
+	require.NoError(t, err)
+
+	// Preparing the SyncML unenroll request
+	requestBytes, err := s.newSyncMLUnenrollMsg(deviceID, targetEndpointURL)
+	require.NoError(t, err)
+
+	resp := s.DoRaw("POST", targetEndpointURL, requestBytes, http.StatusOK)
+
+	// Checking that Command error code was updated
+
+	// Checking response headers
+	require.Contains(t, resp.Header["Content-Type"], microsoft_mdm.SyncMLContentType)
+
+	// Read response data
+	resBytes, err := io.ReadAll(resp.Body)
+	require.NoError(t, err)
+
+	// Checking if response can be unmarshalled to an golang type
+	var xmlType interface{}
+	err = xml.Unmarshal(resBytes, &xmlType)
+	require.NoError(t, err)
+
+	// Checking if device was unenrolled
+	_, err = s.ds.MDMWindowsGetEnrolledDeviceWithDeviceID(context.Background(), deviceID)
+	require.Error(t, err)
```

Comment:
This should be a not found error, right? You should probably assert that the error is that type (I think you can check if the error implements an `interface { IsNotFound() bool }` or more simply - if a bit less reliably - that the error message contains "was not found").